### ISSUE TITLE
Fix the build on non-Windows platforms

### ISSUE THREAD

**Issue by cesarbs** [aspnet/EntityFramework#3594]
Related to https://github.com/aspnet/Universe/issues/304. EntityFramework is currently disabled because we're not able to clone and build it on Linux/OS X to then run tests on CoreCLR.

**Comment by natemcmaster** [comment]
Multiple issues

1. `mono nuget.exe` dies on 4.2.0 (necessary for build.sh)
2. We need to build against Mono 4.2.0 or newer
3. `dnu build` dies on projects with `net46`

**Comment by natemcmaster** [comment]
Blocked on https://github.com/dotnet/corefx/issues/4273

**Comment by natemcmaster** [comment]
Blocked on Mono 4.2.1 bug https://bugzilla.xamarin.com/show_bug.cgi?id=35906

**Comment by muratg** [comment]
Does this work against CoreCLR?  I'm more worried about CoreCLR compat (although Mono would be also nice to have.)

**Comment by natemcmaster** [comment]
I wrote a script for our Travis CI [see travis-build.sh](https://github.com/natemcmaster/EntityFramework/blob/linux-tests/tools/travis-build.sh) that builds and runs only on CoreCLR on Linux. Most of our unit testing appears to work fine. Travis doesn't run SQL Server e2e testing though. I'm working on getting a Linux agent that will test SQL Server connectivity using CoreCLR automatically. It's been so broken up to this point that I've just been manual testing it.

**Comment by natemcmaster** [comment]
All tests now run x-plat on CoreCLR (see #3781). The project will build against Mono, but testing will never fully work unless Mono updates their implementation of SqlClient.

**Comment by natemcmaster** [comment]
Blocked on https://github.com/xunit/xunit/issues/564. Xunit always hangs even if parallelization is turned off.

**Comment by divega** [comment]
@natemcmaster FYI I have updated your check list to include enabling tests in CI on OSX and separated the Mono-specific issues which I think aren't critical to fix.

**Comment by natemcmaster** [comment]
Mostly fixed with fdecc52feb91b42ea7159de5a461526bf589c9a3 which skips failing tests on version of Mono <4.2.0. (Mostly b/c there are some Xunit issues with 4.0, but these are external to EF)

**Comment by divega** [comment]
@natemcmaster Unless it is already completed we need an active bug to track our part of the work for 

- [ ] Need to  Need to enable tests on CoreCLR in CI under OSX (waiting for aspnet/Universe#304)

**Comment by natemcmaster** [comment]
Let's just reuse this bug then.

**Comment by muratg** [comment]
:+1:

**Comment by cesarbs** [comment]
Tests are enabled and running on the CI, but we have that not-yet-reproed failure on `Microsoft.Data.Entity.InMemory.FunctionalTests.QueryInMemoryTest.Where_string_to_lower`.

**Comment by natemcmaster** [comment]
I think the test failure warrants further investigation (e.g. open a new bug.) Current guess: check if KoreBuild/Universe is modifying thread culture.

**Comment by cesarbs** [comment]
https://github.com/aspnet/EntityFramework/issues/4110
